libs/pageserver_api/src/key.rs

```diff
@@ -30,24 +29,25 @@ pub const KEY_SIZE: usize = 18;
 /// See [`Key::to_i128`] for more information on the encoding.
 pub const METADATA_KEY_SIZE: usize = 16;
 
-/// The key prefix start range for the metadata keys. All keys with the first byte >= 0x80 is a metadata key.
-pub const METADATA_KEY_BEGIN_PREFIX: u8 = 0x80;
+/// The key prefix start range for the metadata keys. All keys with the first byte >= 0x40 is a metadata key.
+pub const METADATA_KEY_BEGIN_PREFIX: u8 = 0x40;
```

Comment:
Why should 75% of our keyspace be metadata? Already 50% seems too much to me: most of our keys should be actual data. I think at the least it should not be extended even further.

Reply:
we cannot use >= 0x7F at this point, so it's still 50%.

Reply:
A detailed description of the changes are part of https://github.com/neondatabase/neon/pull/7502, I included that pull request in this pull request to make it pass the tests.